pkgs/development/python-modules/protonvpn-nm-lib/default.nix

```diff
@@ -33,7 +35,14 @@ buildPythonPackage rec {
     systemd
   ];
 
-  # Project has a dummy test.
+  patches = [
+    (substituteAll {
+      src = ./0001-Patching-GIRepository.patch;
+      networkmanager_path = "${networkmanager}/lib/girepository-1.0";
+    })
+  ];
+
+  # Tests can't find systemctl and are not able to access DBus
```

Comment:
Do you know an example of a package doing this kind of testing? I don't know where to start

Reply:
Another thing to consider: To enable tests for this package besides the DBus issue, we need the `systemd-python` library (currently named `systemd` on [nixpkgs](https://github.com/NixOS/nixpkgs/blob/nixos-21.11/pkgs/development/python-modules/systemd/default.nix#L20)) and also the systemd package, because it looks for `systemctl`.

Is it possible to import both without importing the whole `pkgs` to get `systemd` or would it be the best to rename the Python package into something different (although the problem would be fixing the dependent packages, I guess)?

Reply:
e.g. https://github.com/NixOS/nixpkgs/blob/master/nixos/tests/traefik.nix



> Is it possible to import both without importing the whole `pkgs` to get `systemd` or would it be the best to rename the Python package into something different (although the problem would be fixing the dependent packages, I guess)?

You could just input it as a different name 

`` callPackages ../xxx/xxx { python-systemd = systemd; }``